### PR TITLE
OSX build with macos_arch flag support

### PR DESCRIPTION
**Pull request by Antokolos**
Hi, I've tried to build OSX native library and it was necessary to make some changes:

1. When building cpp-bindings there were a lot of "unsupported architecture" error messages. Then I've discovered that the reason for this is my old hardware (Mac with Intel x64 CPU). The solution for this problem is described [here](https://github.com/godotengine/godot-docs/issues/5396).
The solution is adding `macos_arch=x86_64` to the build command to build Intel only binary. Probably it also should be mentioned in the hterrain plugin docs.

2. When building hterrain plugin dynamic library, there were also build errors. Error and its resolution is described [here](https://godotengine.org/qa/65616/problems-compiling-gdnative-c-example-on-osx). Also I've added `macos_arch=x86_64` flag support too (because the name of cpp-bindings depends on macos arch)

The library was built on macOS High Sierra (10.13.6) on MacBook Pro (13-inch, Late 2011). OMG, 12 years old notebook :)

I hope it will be useful, thank you for your work with the plugin, it helped a lot!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -45,6 +45,9 @@ default_env.tres
 # Windows
 *.bat
 
+# OSX
+*.os
```
Comment (Zylann) [inline .gitignore:49]: What is this extension? Last time I saw it, it was actually a very old typo, which should have been `.so` or `.o`, referring to object files during the build.

Comment (Antokolos) [reply .gitignore:49]: Looks like it some kind of object file, these files were created after build:
```
bash-3.2$ ls
gd_library.cpp
gd_library.os
image_utils.cpp
image_utils.os
...

bash-3.2$ file gd_library.os 
gd_library.os: Mach-O 64-bit object x86_64
```

Comment (Zylann) [reply .gitignore:49]: I'm prettty sure it's a typo in GDNative's build files. It should have been `.o`

Comment (Antokolos) [reply .gitignore:49]: Yeah, probably you are right. Probably .gitignore should not be changed then. Unfortunately I cannot find why this files are named this way. But I followed the build instructions :)

Comment (Antokolos) [reply .gitignore:49]: In this .gitignore *.os is listed in section # Compiled source #
https://github.com/2shady4u/godot-sqlite/blob/master/.gitignore
but probably it is a typo in GDNative's build file too, which was copied across the repositories.

**Comment by Zylann** [comment]
I think for the library to be used you also need to modify `hterrain.gdnlib`, and ideally there should be a Github Action to build it for Mac OS.

**Comment by Antokolos** [comment]
It's done, please take a look.
Don't really understood where Github Actions built artifacts are stored, Actions were started after the push to master and finished successfully, but I cannot find where to download built libraries.

**Comment by Zylann** [comment]
On the Windows job for example, it is at the bottom: https://github.com/Zylann/godot_heightmap_plugin/actions/runs/3423983281

Ah also you need to modify `factory.gd` to allow `OSX`, otherwise the plugin won't attempt to load it. Did it work for you?

**Comment by Antokolos** [comment]
Oh, thanks for reminding! I've completely forgot about factory.gd!
Now I've double-checked that library actually loads on my machine (OS name == 'OSX', OS supports arch 'x86_64', therefore is_native_available() returns true)
Also I've downloaded built library and replaced it in repo. I just didn't scrolled to the bottom first time :)
It should be noted though, that the library version built with CI won't work on my machine because it is very old Mac. That's why I've needed `macos_arch=x86_64` switch when building. By default `macos_arch` is `universal` and it should produce universal binary that should work both on Intel and M1 architectures, but unfortunately I cannot test it.

**Comment by Zylann** [comment]
Thanks!

**Pull request merged**